### PR TITLE
French translation

### PR DESCRIPTION
**Pull request by LoubiTek**
## Hello !

I come to propose my French translation.
As well as the favicon optimization that I converted with XnView!

**About**
I should be able to test and try my translation in the browser to see the result and correct in detail. Do not hesitate to tell me if I have make mistakes. Especially for the procedures? I will take the time to check.

Other things, if need I can create a better favicon and / or logo for the site.
And then translat more occasionally of course.

### CODE REVIEW

File: l10n/lang-fr.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      "extras-download-drawing": "Télécharger l'image",
+      "extras-clear-history": "Effacer l'historique",
+      "extras-clear-library": "Effacer la bibliothèque"
+    }
+  },
+  "interpreter": {
+	"messages": {
+      "Can't apply {_PROC_} to special {name:U}": "Ne peut pas appliquer {_PROC_} pour spécial {nom:U",
+      "Can't {_PROC_} primitives unless REDEFP is TRUE": "Ne peut pas {_PROC_} primitives à moins que REDEFP est VRAI",
+      "Can't {_PROC_} special {name:U}": "Ne peut pas {_PROC_} spécial {nom:U}",
```
Comment (inexorabletash) [inline l10n/lang-fr.json]: Don't localize anything in `{}`.

File: l10n/lang-fr.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      "Couldn't parse: '{string}'": "Impossible d'analyser: '{chaîne}'",
```
Comment (inexorabletash) [inline l10n/lang-fr.json]: Don't localize anything in `{}`.

File: l10n/lang-fr.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      "Division by zero": "Division par zéro",
+      "Don't know about variable {name:U}": "Je ne sais pas sur la variable {nom: U}",
```
Comment (inexorabletash) [inline l10n/lang-fr.json]: Don't localize anything in `{}`.

File: l10n/lang-fr.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      "Don't know how to {name:U}": "Je ne sais pas comment {nom: U}",
```
Comment (inexorabletash) [inline l10n/lang-fr.json]: Don't localize anything in `{}`.

File: l10n/lang-fr.json [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      "Don't know what to do with {result}": "Je ne sais pas quoi faire avec {résultat}",
```
Comment (inexorabletash) [inline l10n/lang-fr.json]: Don't localize anything in `{}`.

File: l10n/lang-fr.json [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      "Expected ')'": "Attendu ')'",
+      "Expected ')', saw {word}": "Attendu ')', voir {mot}",
```
Comment (inexorabletash) [inline l10n/lang-fr.json]: Don't localize anything in `{}`.

File: l10n/lang-fr.json [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+      "Expected string": "Chaîne attendue",
+      "No output from procedure": "Pas de sortie de procédure",
+      "TO: Expected END":  "FIN attendue",
+      "TO: Expected identifier": "Identifiant attendu",
+      "Unexpected '{c}'": "Inattendu '{c}'",
+      "Unexpected end of instructions": "Fin inattendue des instructions",
+      "{_PROC_}: Array size must be positive integer": "{_PROC_}: La taille du tableau doit être un nombre entier positif",
+      "{_PROC_}: Can't create circular array": "{_PROC_}: Impossible de créer un tableau circulaire",
+      "{_PROC_}: Can't overwrite primitives unless REDEFP is TRUE": "{_PROC_}: Impossible d'écraser les primitives à moins que REDEFP ne soit VRAI",
+      "{_PROC_}: Can't overwrite special {name:U}": "{_PROC_}: Impossible d'écraser {nom: U}",
```
Comment (inexorabletash) [inline l10n/lang-fr.json]: Don't localize anything in `{}`.

File: l10n/lang-fr.json [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+      "{_PROC_}: Can't redefine primitive {name:U}": "{_PROC_}: Impossible de redéfinir la primitive {nom: U}",
```
Comment (inexorabletash) [inline l10n/lang-fr.json]: Don't localize anything in `{}`.

File: l10n/lang-fr.json [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+      "{_PROC_}: Can't show definition of primitive {name:U}": "{_PROC_}: Impossible d'afficher la définition de la primitive {nom: U}",
```
Comment (inexorabletash) [inline l10n/lang-fr.json]: Don't localize anything in `{}`.

File: l10n/lang-fr.json [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+      "sb-link-text-links": "autres ressources de Logo",
+      "extras-download-library": "Télécharger la bibliothèque",
+      "extras-download-drawing": "Télécharger l'image",
+      "extras-clear-history": "Effacer l'historique",
+      "extras-clear-library": "Effacer la bibliothèque"
+    }
+  },
+  "interpreter": {
+	"messages": {
+      "Can't apply {_PROC_} to special {name:U}": "Ne peut pas appliquer {_PROC_} pour spécial {nom:U",
```
Comment (inexorabletash) [inline l10n/lang-fr.json]: Don't localize anything in {}. That is, do not change {name:U} to {nom:U}. The name is a reference to an internal variable.

**Comment by LoubiTek** [comment]
I just made the corrections.

By cons, I do not know how to go back on a commit..
So if you ever have the opportunity to do something ?!
So that I can be restart the patch separately for the favicon.

**Comment by inexorabletash** [comment]
FYI, to go back on a commit you'd just check out an older version, e.g.:
```
git checkout 6d23d6a29a9a9f8a72ec3e30314480d0d1985126 favicon.ico
```
It's okay though, I see the comment now that it's just an optimized image which is why I didn't see a difference.

**Pull request merged**